src/spesh/arg_guard.c

```diff
@@ -647,7 +653,9 @@ void MVM_spesh_arg_guard_destroy(MVMThreadContext *tc, MVMSpeshArgGuard *ag, MVM
 void MVM_spesh_arg_guard_discard(MVMThreadContext *tc, MVMStaticFrame *sf) {
     MVMStaticFrameSpesh *spesh = sf->body.spesh;
     if (spesh && spesh->body.spesh_arg_guard) {
-        MVM_spesh_arg_guard_destroy(tc, spesh->body.spesh_arg_guard, 1);
-        spesh->body.spesh_arg_guard = NULL;
+        /* No need to atomically exchange with NULL here, since actual freeing
+         * of data is postponed to the next safe point. */
+        MVM_spesh_arg_guard_destroy(tc, (MVMSpeshArgGuard *)MVM_load(&spesh->body.spesh_arg_guard), 1);
+        MVM_store(spesh->body.spesh_arg_guard, NULL);
```

Comment:
If there's "No need to atomically exchange with NULL here ...", then why the `MVM_store(spesh->body.spesh_arg_guard, NULL);` instead of just `spesh->body.spesh_arg_guard = NULL;`?

Reply:
removed the comment, it wasn't really helpful